src/pyperclip/__init__.py

```diff
@@ -529,6 +562,11 @@ def determine_clipboard():
         else:
             return init_gtk_clipboard()
 
+        if (
+                os.environ["XDG_SESSION_TYPE"] == "wayland" and
```

Comment:
This isn't a good environment variable to check. For example, I run sway as my compositor and login via tty. This variable is set as `tty` for me and thus fails the this check. I would suggest simply `os.environ["WAYLAND_DISPLAY"] != ""` instead here. Every compositor will set that one to something (usually `wayland-0`) and no xorg WM will ever set that variable, so it should be the safest thing to check.